front/app/hooks/useEvents.ts

```diff
@@ -1,44 +1,68 @@
 import { useState, useEffect } from 'react';
 import { isNilOrError } from 'utils/helperUtils';
-import {
-  IEventData,
-  eventsStream,
-  IProjectsStreamParams,
-} from 'services/events';
-
-export default function useEvents(
-  projectIds?: string[],
-  futureOnly?: boolean,
-  pastOnly?: boolean
-) {
+import { getPageNumberFromUrl } from 'utils/paginationUtils';
+import { IEventData, eventsStream, IEventsStreamParams } from 'services/events';
+
+interface InputParameters {
+  projectIds?: string[];
+  futureOnly?: boolean;
+  pastOnly?: boolean;
+  pageNumber?: number;
+  pageSize?: number;
+}
+
+export default function useEvents({
+  projectIds,
+  futureOnly,
+  pastOnly,
+  pageNumber,
+  pageSize,
+}: InputParameters) {
   const [events, setEvents] = useState<IEventData[] | undefined | null | Error>(
     undefined
   );
+  const [lastPageNumber, setLastPageNumber] = useState(1);
 
   useEffect(() => {
     setEvents(undefined);
 
-    const streamParams: IProjectsStreamParams = {
+    const streamParams: IEventsStreamParams = {
       queryParameters: { project_ids: projectIds },
     };
 
     if (futureOnly) {
-      streamParams.queryParameters.start_at_gteq = new Date();
+      streamParams.queryParameters.start_at_gteq = new Date().toJSON();
     }
 
     if (pastOnly) {
-      streamParams.queryParameters.start_at_lt = new Date();
+      streamParams.queryParameters.start_at_lt = new Date().toJSON();
+    }
+
+    if (pageNumber) {
+      streamParams.queryParameters['page[number]'] = pageNumber;
+    }
+
+    if (pageSize) {
+      streamParams.queryParameters['page[size]'] = pageSize;
     }
 
     const subscription = eventsStream(streamParams).observable.subscribe(
       (response) => {
-        const events = !isNilOrError(response) ? response.data : response;
-        setEvents(events);
+        if (isNilOrError(response)) {
+          setEvents(response);
+          setLastPageNumber(1);
+          return;
+        }
+
+        setEvents(response.data);
+
+        const lastPageNumber = getPageNumberFromUrl(response.links?.last) ?? 1;
```

Comment:
Personal preference (so feel free to ignore): keep const grouped at the top and setState/function calls grouped at the bottom (as in useModerations)